compiler/packages/babel-plugin-react-compiler/src/Inference/InferReferenceEffects.ts

```diff
@@ -1330,7 +1344,8 @@ function inferBlock(
             state.referenceAndRecordEffects(
               freezeActions,
               place,
-              Effect.Read,
+              // see call-spread-argument-mutable-iterator test fixture
+              arg.kind === 'Spread' ? Effect.ConditionallyMutate : Effect.Read,
```

Comment:
hmm this doesn't seem like it should be necessary - we're inside an `if (areArgumentsImmutableAndNonMutating())` check